src/bootstrap/bootstrap.py

```diff
@@ -12,9 +12,25 @@
 import sys
 import tarfile
 import tempfile
+import sqlite3
 
 from time import time
 
+def acquire_lock():
+    try:
+        con = sqlite3.Connection("xlock.db", timeout=0)
+        curs = con.cursor()
+        curs.execute("BEGIN EXCLUSIVE")
+        return curs
+    except sqlite3.OperationalError:
+        del con
+        del curs
+        print("Waiting for lock on database")
+        con = sqlite3.Connection("xlock.db", timeout=9999)
+        curs = con.cursor()
+        curs.execute("BEGIN EXCLUSIVE")
```

Comment:
Does this need to do some sort of cleanup to release the lock? Or is exiting the process enough?

Reply:
Dropping the cursor should automatically end the transaction.

Reply:
Have we verified what happens when x.py is killed (kill -9, or similar)? I would expect that the lock is just indefinitely held?